logbook-api/src/main/java/org/zalando/logbook/DefaultHttpHeaders.java

```diff
@@ -43,34 +43,43 @@ public HttpHeaders update(
             final String name,
             final Collection<String> values) {
 
-        return withHeaders(
-                headers.assoc(name, immutableCopy(values)));
+        TreeMap<String, List<String>> updatedHeaders = associate(headers, name, values);
+        return withHeaders(updatedHeaders);
     }
 
     @Override
     public HttpHeaders delete(final Collection<String> names) {
-        return withHeaders(
-                Fold.fold(names, headers, ImSortedMap::without));
+        return withHeaders(delete(headers, names));
     }
 
-    public static <T> List<T> immutableCopy(final Collection<T> values) {
-        if (values instanceof ImListWithToString) {
-            return (List<T>) values;
-        }
-        return new ImListWithToString<>(ofIter(values));
+    public static List<String> immutableCopy(final Collection<String> values) {
+        return Collections.unmodifiableList(new ArrayList<>(values));
     }
 
-    @AllArgsConstructor
-    private static final class ImListWithToString<T> extends AbstractList<T> {
+    private static TreeMap<String, List<String>> associate(
+            final TreeMap<String, List<String>> original,
+            final String key,
+            final Collection<String> value) {
 
-        private final ImList<T> list;
+        if (Objects.equals(original.get(key), value))
+            return original;
 
-        @Delegate
-        @SuppressWarnings("unused")
-        private List<T> delegate() {
-            return list;
-        }
+        TreeMap<String, List<String>> tmpMap = new TreeMap<>(original);
+        tmpMap.put(key, immutableCopy(value));
+        return tmpMap;
+    }
+
+    private static TreeMap<String, List<String>> delete(
+            final TreeMap<String, List<String>> original,
+            final Collection<String> keys) {
+
+        if (Collections.disjoint(original.keySet(), keys))
+            return original;
 
+        Comparator<String> cmp = String.CASE_INSENSITIVE_ORDER;
+        TreeMap<String, List<String>> modifiedMap = new TreeMap<>(original);
+        modifiedMap.keySet().removeIf(k -> keys.stream().anyMatch(key -> cmp.compare(k, key) == 0));
+        return modifiedMap;
```

Comment:
Sorry, maybe it's just me, but wouldn't it be simpler to have something like this? The whole direct comparator usage seems unnecessary to my taste.
```suggestion
        TreeMap<String, List<String>> modifiedMap = new TreeMap<>(original);
        keys.forEach(modifiedMap::remove);
        return modifiedMap;
```

Reply:
Thanks, this is a great suggestion - The simpler, the better!
I applied it.